sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/resource/project/ProjectResourceService.groovy

```diff
@@ -0,0 +1,34 @@
+package life.qbic.portal.sampletracking.resource.project
+
+import life.qbic.datamodel.dtos.projectmanagement.Project
+import life.qbic.portal.sampletracking.resource.ResourceService
+
+/**
+ * <p>This resource service holds projects.</p>
+ * *
+ * @since 1.0.0
+ */
+class ProjectResourceService extends ResourceService<Project> {
+
+    @Override
+    <RESOURCE extends Project> void addToResource(RESOURCE resourceItem) {
+        content.add(resourceItem)
+        ProjectAddedMessage projectAddedMessage = new ProjectAddedMessage(resourceItem)
+        publish(projectAddedMessage)
```

Comment:
How about, instead of creating message type objects, sending the message type through the publish method?

I see the issue that we create lots and lots of message objects. They do not provide other functionality than having the message type. And if I see it correctly you do not enforce the use of the message type directly. With doing it through the publishe method you would still know the message type and you have less overhead with creating new services

Reply:
I introduced Topics now to do that. Thank you for your valuable feedback. I think it helped a lot in simplifying this.